From 58eb9f25bdac2d4553f339d1c6a9b1430751378f Mon Sep 17 00:00:00 2001
From: Richard Liu <r66033@freescale.com>
Date: Tue, 16 Jul 2013 11:20:12 +0800
Subject: [PATCH 2/3] ENGR00276102 fix Music pop up stop issue

Music pop up Unfortunately music has stopped
when switch between shuffle and repeat.

Signed-off-by: Richard Liu <r66033@freescale.com>
---
 src/com/android/music/MediaPlaybackService.java | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/com/android/music/MediaPlaybackService.java b/src/com/android/music/MediaPlaybackService.java
index d25e929..c5cc962 100644
--- a/src/com/android/music/MediaPlaybackService.java
+++ b/src/com/android/music/MediaPlaybackService.java
@@ -1085,8 +1085,15 @@ public class MediaPlaybackService extends Service {
     }
 
     private void setNextTrack() {
-        mNextPlayPos = getNextPosition(false);
-        if (mNextPlayPos >= 0) {
+       int pos = getNextPosition(false);
+        if (pos < 0) {
+            gotoIdleState();
+            if (mIsSupposedToBePlaying) {
+               mIsSupposedToBePlaying = false;
+               notifyChange(PLAYSTATE_CHANGED);
+            }
+        } else {
+            mNextPlayPos = pos;
             long id = mPlayList[mNextPlayPos];
             mPlayer.setNextDataSource(MediaStore.Audio.Media.EXTERNAL_CONTENT_URI + "/" + id);
         }
-- 
1.8.0

